pkg/proxy/iptables/proxier.go

```diff
@@ -1220,6 +1294,14 @@ func (proxier *Proxier) syncProxyRules() {
 				// Error parsing this endpoint has been logged. Skip to next endpoint.
 				continue
 			}
+
+			// Only use topologically-best set of endpoints if it's been filtered.
+			if filteredEndpoints != nil {
```

Comment:
add feature gate check here as well?

Reply:
If geature gate is not enabled, `filteredEndpoints` will always be nil, so I think we can just remove feature gate check at line 1256 ?